eisbuk-admin/src/components/temp/NewSlotForm/SelectCategories.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from "react";
+import { useField, ErrorMessage } from "formik";
+import { useTranslation } from "react-i18next";
+
+import Box from "@material-ui/core/Box";
+import Checkbox from "@material-ui/core/Checkbox";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+import { SlotType, Category } from "eisbuk-shared";
+
+import { categoryLabel } from "@/lib/labels";
+// #region main
+
+/**
+ * Checkboxes selection field for categories.
+ * Registers `"categories"` field to form, handles updates and errors.
+ * Disabled (with all categories selected) if value for `type` (in form context)
+ * `"off_ice_dancing" | "off_ice_gym"`
+ */
+const SelectCategories: React.FC = () => {
+  const classes = useStyles();
+
+  const [disabled, setDisabled] = useState(false);
+
+  // check value for "type" field in the form context
+  const [{ value: slotType }] = useField<SlotType>("type");
+  // we want to set value for "categories" to all if "type" is off_ice
+  const [, , { setValue }] = useField<string[]>("categories");
+
+  useEffect(() => {
+    // if slot type off_ice, disable all checkboxes and set value for "categories" to all
+    if ([SlotType.OffIceDancing, SlotType.OffIceGym].includes(slotType)) {
+      setValue(Object.values(Category));
+
+      setDisabled(true);
+    } else {
+      setDisabled(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [slotType]);
```

Comment:
Shouldn't the dependencies array include all state dependent value used inside the clojure?

I expected this list to be `[slotType, setValue, setDisabled]`

Reply:
This is intentional as it was causing infinite loops.